Add a Macros chapter (#137)
apple/swift-book: TSPL.docc/ReferenceManual/Attributes.md

You can apply a declaration attribute to declarations only.

### attached

> Apply the `attached` attribute, to the declaration of an macro.

Reviewer: ```suggestion
Apply the `attached` attribute, to the declaration of a macro.
```

Reviewer: Or should that be "any"?

- Member macros have `member` as the first argument to this attribute.
  The type that implements the macro conforms to the `MemberMacro` protocol.
  These macros produce new declarations
  that are members of the type or extension
  that the macro is attached to.
  For example,
  applying a member macro to a structure declaration
  can define additional methods and properties on that structure.

- Member attribute macros have `memberAttribute` as the first argument to this attribute.
  The type that implements the macro conforms to the `MemberAttributeMacro` protocol.
  These macros add attributes to members of the type or extension
  that the macro is attached to.

- Accessor macros have `accessor` as the first argument to this attribute.
  The type that implements the macro conforms to the `AccessorMacro` protocol.
  These macros add accessors to the stored property they're attached to,
  turning it into a computed property.

- Conformance macros have `conformance` as the first argument to this attribute.
  The type that implements the macro conforms to the `ConformanceMacro` protocol.
  These macros add protocol conformance to the type they're attached to.

> The XXX and XXX macro roles require a `named:` argument,

Reviewer: FIXME

listing the names of the symbols that the macro generates.
When a macro declaration includes the `named:` argument,
the macro implementation must generate
only symbol with names that match that list.
That said,
a macro need not generate a symbol for every listed name.
The value for that argument is a list of one or more of the following:

- `named(<#name#>)`
  where *name* is that fixed symbol name,

>   for a name that's known in advanced

Reviewer: ```suggestion
  for a name that's known in advance.
```

- `overloaded`

>   for a name that's the same as an existing symbol

Reviewer: ```suggestion
  for a name that's the same as an existing symbol.
```

- `prefixed(<#prefix#>)`
  where *prefix* is prepended to the symbol name,

>   for a name that starts with a fixed string

Reviewer: ```suggestion
  for a name that starts with a fixed string.
```

>   for a name that starts with a fixed string

Reviewer: ```suggestion
  for a name that starts with a fixed string.
```

- `suffixed(<#suffix#>`
  where *suffix* is appended to the symbol name,

>   for a name that ends with a fixed string

Reviewer: ```suggestion
  for a name that ends with a fixed string.
```

> - `arbirtary`

Reviewer: ```suggestion
- `arbitrary`
```

>   for a name that can't be determined until the macro is expanded

Reviewer: ```suggestion
  for a name that can't be determined until the macro expands.
```
